src/pharmpy/tools/modelsearch/tool.py

```diff
@@ -111,7 +99,64 @@ def post_process(rank_type, cutoff, *models):
     return res
 
 
-class ModelSearchResults(pharmpy.results.Results):
+@same_signature_as(create_workflow)
+def validate_input(
+    search_space,
+    algorithm,
+    iiv_strategy,
+    rank_type,
+    cutoff,
+    model,
+):
+
+    if not hasattr(algorithms, algorithm):
+        raise ValueError(
+            f'Invalid algorithm: got "{algorithm}" of type {type(algorithm)},'
+            f' must be one of {sorted(dir(algorithms))}.'
+        )
+
+    if rank_type not in RANK_TYPES:
+        raise ValueError(
+            f'Invalid rank_type: got "{rank_type}" of type {type(rank_type)},'
+            f' must be one of {sorted(RANK_TYPES)}.'
+        )
+
+    if iiv_strategy not in algorithms.IIV_STRATEGIES:
+        raise ValueError(
+            f'Invalid IIV strategy: got "{iiv_strategy}" of type {type(iiv_strategy)},'
+            f' must be one of {sorted(algorithms.IIV_STRATEGIES)}.'
+        )
+
+    if not isinstance(cutoff, (type(None), int, float)):
+        raise TypeError(
+            f'Invalid cutoff: got "{cutoff}" of type {type(cutoff)},'
+            f' must be None/NULL or an int/float.'
+        )
+
+    try:
+        parse(search_space)
+    except:  # noqa E722
+        raise ValueError(f'Invalid search_space, could not be parsed: "{search_space}"')
+
+    validate_model(model)
+
+
+def validate_model(model):
```

Comment:
I think in this particular case here it's just because we do not have the arguments in scope to check them in the `start` task. But checking the input in the start task of the workflow should probably be the default for all tools if we ever want to compose their workflows. Right now the `validate_model` could be skipped since I do not think we compose any of those workflows.

Reply:
Got rid of `validate_model` for now.